File: android/src/main/java/com/reactlibrary/IsMutedModule.java
package com.nielsotten.rnismuted;

import android.content.Context;
import android.media.AudioManager;
import com.facebook.react.bridge.ReactApplicationContext;
import com.facebook.react.bridge.ReactContextBaseJavaModule;
import com.facebook.react.bridge.ReactMethod;
import com.facebook.react.bridge.Promise;

public class IsMutedModule extends ReactContextBaseJavaModule {

    private final ReactApplicationContext context;

    public IsMutedModule(ReactApplicationContext context) {
        super(context);
        this.context = context;
    }

    @Override
    public String getName() {
        return "IsMuted";
    }

    @ReactMethod
    public void isMuted(Promise promise) {
        try {
            AudioManager audioManager = (AudioManager) context.getSystemService(Context.AUDIO_SERVICE);
            if (audioManager.getRingerMode() == AudioManager.RINGER_MODE_NORMAL)
                promise.resolve(false);
            else
                promise.resolve(true);
        } catch (Exception exception) {
            promise.reject(exception);
        }
    }
}

File: android/src/main/java/expo/modules/ismuted/IsMutedModule.kt
package expo.modules.ismuted

import android.content.Context
import android.media.AudioManager
import expo.modules.kotlin.exception.CodedException
import expo.modules.kotlin.modules.Module
import expo.modules.kotlin.modules.ModuleDefinition

class IsMutedModule : Module() {
  override fun definition() = ModuleDefinition {
    Name("IsMuted")

    AsyncFunction("isMuted") {
      val context = appContext.reactContext
        ?: throw CodedException("ERR_IS_MUTED", "React context is unavailable.", null)
      val audioManager = context.getSystemService(Context.AUDIO_SERVICE) as AudioManager
      isAudioMuted(audioManager)
    }
  }
}

internal fun isAudioMuted(audioManager: AudioManager): Boolean =
  audioManager.ringerMode != AudioManager.RINGER_MODE_NORMAL
